oomact_ros/src/CalibrationNode.cpp

```diff
@@ -5,98 +5,113 @@
 #include <aslam/calibration/ros/RosInputProvider.h>
 #include <aslam/calibration/calibrator/CalibratorI.hpp>
 #include <aslam/calibration/model/sensors/PoseSensor.hpp>
+#include <aslam/calibration/model/sensors/PositionSensor.hpp>
+
+#include <sm/BoostPropertyTree.hpp>
+#include <sm/value_store/PropertyTreeValueStore.hpp>
 
 namespace cal = aslam::calibration;
 
-int main(int argc, char **argv) {
+std::string getFilePathFromRosParam(const std::string& param_name,
+                                    ros::NodeHandle* nh) {
+  std::string file_path;
+  if (!nh->getParam(param_name, file_path)) {
+    ROS_FATAL_STREAM("Could not find " << file_path << ", exiting");
+    exit(EXIT_FAILURE);
+  }
+  return file_path;
+}
+
+void loadSensorParameters(const cal::ValueStoreRef vs_sensors,
+                          std::shared_ptr<cal::FrameGraphModel> model,
+                          std::vector<std::unique_ptr<cal::Sensor>>* sensors) {
+  DCHECK(sensors != nullptr);
```

Comment:
The output param pointer is a bit of a style alien in this library. This is a very googlish thing. And me personally I don't agree with this at all. I'd make it a non const ref here. That would also remove the runtime check. (not important)